libs/platform/src/lib/components/list/list.component.ts

```diff
@@ -0,0 +1,662 @@
+    _startIndex = 0;
+
+    /** @hidden
+    */
+    _lastIndex = this.itemSize;
+
+    /** @hidden
+    */
+    _dsItems = [];
+
+    /** @hidden
+    */
+    _destroyed = new Subject<void>();
+
+    /**@hidden
+     * for data source handling
+     */
+    _dsSubscription: Subscription | null;
+
+    /**@hidden
+    * for items impertaive and declartive approaches
+    */
+    _itemsSubscription: Subscription | null;
+
+    /**
+    * Child items of the List.
+    */
+    @ContentChildren(BaseListItem, { descendants: true })
+    ListItems: QueryList<BaseListItem>;
+
+    @ContentChild(ListItemDef)
+    listItemDef: ListItemDef;
+
+    /** Access child element, for checking link content*/
+    @ViewChild('link', { read: ElementRef })
+    anchor: ElementRef;
+
+    /** role */
+    @HostBinding('attr.role')
+    role = 'list';
+
+    /** @hidden */
+    @Output()
+    selectedItemChange: EventEmitter<SelectionChangeEvent> = new EventEmitter<SelectionChangeEvent>();
+
+    /**
+    * Datasource for suggestion list
+    */
+    @Input()
+    get dataSource(): FdpListDataSource<any> {
+        return this._dataSource;
+    }
+    set dataSource(value: FdpListDataSource<any>) {
+        if (value) {
+            this._initializeDS(value);
+        }
+    }
+    protected _dataSource: FdpListDataSource<any>;
+
+    /**
+    * content Density of element. 'cozy' | 'compact'
+    */
+    @Input()
+    set contentDensity(contentDensity: ContentDensity) {
+        this._contentDensity = contentDensity;
+        this.isCompact = contentDensity === 'compact';
+    }
+
+    /** setter and getter for _navigated */
+    @Input('navigated')
+    get navigatedValue(): boolean {
+        return this.navigated;
+    }
+
+    set navigatedValue(value: boolean) {
+        this.navigated = value;
+        this.itemEl.nativeElement.querySelector('ul').classList.add('fd-list--navigation');
+    }
+
+    /** setter and getter for _navigationIndicator */
+    @Input('navigationIndicator')
+    get navigationIndicatorValue(): boolean {
+        return this.navigationIndicator;
+    }
+
+    set navigationIndicatorValue(value: boolean) {
+        this.navigationIndicator = value;
+        this.itemEl.nativeElement.querySelector('ul').classList.add('fd-list--navigation-indication');
+    }
+
+    /** setter and getter for hasByLine*/
+    @Input('hasByLine')
+    get hasByLineValue(): boolean {
+        return this.hasByLine;
+    }
+
+    set hasByLineValue(value: boolean) {
+        this.hasByLine = value;
+        this.itemEl.nativeElement.querySelector('ul').classList.add('fd-list--byline');
+    }
+
+    /** setter and getter for radio button and checkbox*/
+    @Input()
+    get value(): any {
+        return super.getValue();
+    }
+
+    set value(value: any) {
+        super.setValue(value);
+    }
+
+
+    /** @hidden */
+    /**handline keyboard operations
+    * in template on list and list items
+    */
+    handleKeyDown(event: KeyboardEvent): boolean {
```

Comment:
These should under methods section under constructors, above constructor we have only properties.

Reply:
done